### ISSUE TITLE
ansible failed if (other?) version of ansible installed on target host

### ISSUE THREAD

**Issue by olfway** [ansible/ansible#16187]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

ansible cannot execute modules on target host, if other version of ansible installed on it
##### STEPS TO REPRODUCE

```
target: sudo pip install ansible
ansible -m ping target
```
##### ACTUAL RESULTS

```
target | FAILED! => {
    "changed": false,
    "failed": true,
    "msg": "Error: Module unable to locate ANSIBLE_MODULE_ARGS and ANSIBLE_MODULE_CONSTANTS in json data from stdin.  Unable to figure out what parameters were passed"
}
```

**Comment by abadger** [comment]
I'm unable to reproduce this.

```
$ ansible --version                 [stable-2.1]  (15:56:23)
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
$ ansible -m ping -i 'fedora22-test,' all
fedora22-test | SUCCESS => {
    "changed": false, 
    "ping": "pong"
```

I've tried with pip installs of ansible==2.1.0, ansible==2.0.2.0, and ansible==1.9.6 on the remote machines.

This is a problem that may have occurred with git checkouts of ansible before 2.1.0 final, can you verify what versions of ansible are present on your controller and targets when you run this?

**Comment by olfway** [comment]
I have ansible 2.1.0.0 release on my controller machine

on the target it says "ansible 2.2.0", i guess it's snapshot from ~ 4 May 2016

**Comment by abadger** [comment]
How is the snapshot installed?

**Comment by olfway** [comment]
i found in bash_history:

```
git clone https://github.com/ansible/ansible.git
cd ansible/
sudo make install
```

git describe in this workdir says:
v2.0.0-0.5.beta3-2194-g56ba103

which ansible -> /usr/local/bin/ansible

ls -l /usr/local/lib/python2.7/dist-packages

```
total 12
drwxr-sr-x 4 root staff 4096 May  4 22:48 ansible-2.2.0-py2.7.egg
-rw-r--r-- 1 root staff  244 May  4 22:17 easy-install.pth
-rw-r--r-- 1 root staff   33 May  4 22:48 setuptools.pth
```

cat easy-install.pth

```
import sys; sys.__plen = len(sys.path)
./ansible-2.2.0-py2.7.egg
/usr/lib/python2.7/dist-packages
import sys; new=sys.path[sys.__plen:]; del sys.path[sys.__plen:]; p=getattr(sys,'__egginsert',0); sys.path[p:p]=new; sys.__egginsert = p+len(new)
```

**Comment by abadger** [comment]
Reproduced by installing the way you specified... I think it has to do with setuptools doing its own path munging :-(  Looking into what we can do about that.

**Comment by abadger** [comment]
Yeah, it's the easy-install.pth that's messing things up.  That .pth file is injecting ansible-2.2.0-py2.7.egg into the path earlier than anything that we've already done.  setuptools is the worst :-(

**Comment by abadger** [comment]
Testing on a similar bug showed that the PR fixed things.  Merged to devel and stable-2.1 and the fix should be in our next 2.1.x release.
